docs/TrafficSplitting.md

```diff
@@ -0,0 +1,120 @@
+# Traffic Splitting
+
+The purpose of this section of the kn documentation is to describe the command verbs for traffic splitting operations and present some examples.
+
+`kn` client can work with following properties of a traffic target defined in traffic block of a service spec.
+ - **RevisionName**: RevisionName of a specific revision to which to send a portion of traffic.
+ - **Tag**: Tag is optionally used to expose a dedicated url for referencing this target exclusively.
+ - **Percent**: Percent specifies percent of the traffic to this Revision.
```

Comment:
Just wanted to point out some key things in case people are unaware:
- The `tag` property on a traffic block is optional.  This means that we CAN NOT identify a traffic block by this value, since not all blocks will have one.
- The `revisionName` property on a traffic block may not be unique. More than one traffic block can refer to the same revisionName. This means that we CAN NOT identify a traffic block by this value.
- Additionally, since people can use `latestRevision: true`, there is no consistent revisionName to search on for this block (it'll change each time a new revision is created) - which requires the user to query the service each time they want to modify this traffic block to see if the value changed.  Otherwise, per this PR, a new block will be create w/o them realizing it (I think).

IMO this means any operation that tries to modify an existing traffic block probably can't because there's no

Reply:
> * The `tag` property on a traffic block is optional.  This means that we CAN NOT identify a traffic block by this value, since not all blocks will have one.

True, therefore a user can use revision or tag  (or both in case of multiple revisions being present). As mentioned above we might need to add a special name to target the "latestRevision" traffic block.

> * The `revisionName` property on a traffic block may not be unique. More than one traffic block can refer to the same revisionName. This means that we CAN NOT identify a traffic block by this value.

But the combination tag:revision is unique. That's how you would need to address such a block.

Reply:
only if tag is present - which it may not be because it's optional.  I think having a different traffic-block-picking syntax based on the data in the traffic sections will be a bit confusing.
We need a cmd that works consistently.

Reply:
> only if tag is present - which it may not be because it's optional.

Yes, and which gives us opportunity to design UX to remove the ambiguities. 

> I think having a different traffic-block-picking syntax based on the data in the traffic sections will be a bit confusing.

yes, and this is why we need to give users a way to not confuse them. We should ask users to create unique identity for ambiguous targets, which is `Tag`. We ask users to create `Tag` for a revision if it has to present more than once in traffic block. So that sub-subsequent operations can be performed without ambiguity. `Tag` is optional, I agree, but we can leverage it for identifying a target uniquely.

There are a bunch of combinations, given that serving traffic block allows fields as optional and aliases and I think we should work towards making the client UX easy to understand and execute for users, and defining the path to reach to a desired state.